docs/agent-reference/agent-core/configuration-manager.md

```diff
@@ -0,0 +1,95 @@
+# Device Update Agent Configuration
+
+## APIs
+
+```c
+// A config object handle.
+typedef void* CONFIG_HANDLE
+
+// Instancitate a config object from specified file.
+CONFIG_HANDLE LoadConfig(const char * configFilePath);
+
+// Free all resources allocated for the config object.
+void FreeConfig(CONFIG_HANDLE config);
+
+// Returns value of the specified config name.
+// (Output is a json string).
+const char* GetValue(const char* name);
+
+```
+
+## Configuration File Format
+
+See example `du-config.json` file format below:
+
+```json
+
+"enabledIoTHubTracing" : true,
+"logLevel" : 0,
+
+"aduShellTrustedUsers": ["adu", "do"],
+
+"agents":[
+  {
+      "name":"host-update",
+      "runas":"adu",
+      "connectionInfoSource":"AIS",
+      "connectionData":{
+          "connectionType" : "module",
+          "moduleName" : "ioTHubDeviceUpdate"
+      },
+
+      "aduc_manufacturer" : "Contoso",
+      "aduc_model" : "Smart-Box"
+  },
+
+  {
+      "name":"leaf-update",
+      "runas":"adu",
+      "connectionInfoSource":"string",
+      "connectionData":{
+          "connectionString" : "HostName=..."
+      },
+
+      "aduc_manufacturer" : "Fabrikam",
+      "aduc_model" : "Camera"
+  }
+],
+
+"contentHandlers":[
+    {
+        "updateType":"microsoft/apt:1",
+        "path":"/user/lib/adu/content-handlers.d/microsoft-apt-1/libmicrosoft-apt-1.so",
+        "sha256":"xAbsdf802x3233="
```

Comment:
> "sha256":"xAbsdf802x3233=" [](start = 7, length = 27)

sha should be supported inclusively.

e.g. support sha256+ or whatever we decide is the minimum value

Also deserves a note that all of our shas are not pure SHA values. They are Base64 encodings of Sha(whatever) hashes.

Reply:
@nihemstr , noted. thx.